app/components/interactive_dialog_controller/interactive_dialog_controller.js

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+
+export default class InteractiveDialogController extends PureComponent {
+    static propTypes = {
+        triggerId: PropTypes.string,
+        dialog: PropTypes.object,
+        navigator: PropTypes.object,
+        theme: PropTypes.object,
+    };
+
+    componentDidUpdate(prevProps) {
+        const triggerId = this.props.triggerId;
+        if (prevProps.triggerId === triggerId) {
+            return;
+        }
+
+        const dialogData = this.props.dialog;
+        if (dialogData.trigger_id !== triggerId) {
+            return;
+        }
+
+        if (!dialogData.trigger_id || !dialogData.dialog) {
+            return;
+        }
+
+        const theme = this.props.theme;
+
+        this.props.navigator.push({
```

Comment:
are we confident about this? by looking at the invision this screen should be a modal, also have you tested this everywhere where we show posts? (channel screen, thread, recent mentions, flagged posts, search results, permalinks,etc...)

Reply:
I really don't understand the difference between screens and modals. When should I use one or the other and why does it matter?

Reply:
A modal is a screen that appears on top of all the others, creating a new stack, a regular screen will just put itself on top of the stack.

A modal is normally used when you need to offer the ability to accomplish a self-contained task related to your app’s primary function. A modal view is especially appropriate for a multistep subtask that requires UI elements that don’t belong in the main app user interface all the time.

those this help?

Reply:
A little, but what are the functional differences? Does creating a new stack give any benefit over pushing to the existing stack?

Reply:
the difference is the way the app behaves and transitions screens (more noticeable on iOS) and how we handle the stack of screens

Reply:
Ok, should I switch to using a modal then?

Reply:
and you'll see how it does not include a back button as default and you need to set a `leftButton` to handle close/cancel

Reply:
yes please do switch